lib/src/chewie_controller.dart

```diff
@@ -0,0 +1,182 @@
+    this.cupertinoProgressColors,
+    this.materialProgressColors,
+    this.placeholder,
+    this.showControls = true,
+    this.allowedScreenSleep = true,
+    this.isLive = false,
+  })  : assert(videoPlayerController != null,
+            'You must provide a controller to play a video'),
+        super(ChewieValue(videoPlayerController)) {
+    _initialize();
+  }
+
+  /// Initialize the Video on Startup. This will prep the video for playback.
+  final bool autoInitialize;
+
+  /// Play the video as soon as it's displayed
+  final bool autoPlay;
+
+  /// Start video at a certain position
+  final Duration startAt;
+
+  /// Whether or not the video should loop
+  final bool looping;
+
+  /// Whether or not to show the controls
+  final bool showControls;
+
+  /// The Aspect Ratio of the Video. Important to get the correct size of the
+  /// video!
+  ///
+  /// Will fallback to fitting within the space allowed.
+  final double aspectRatio;
+
+  /// The colors to use for controls on iOS. By default, the iOS player uses
+  /// colors sampled from the original iOS 11 designs.
+  final ChewieProgressColors cupertinoProgressColors;
+
+  /// The colors to use for the Material Progress Bar. By default, the Material
+  /// player uses the colors from your Theme.
+  final ChewieProgressColors materialProgressColors;
+
+  /// The placeholder is displayed underneath the Video before it is initialized
+  /// or played.
+  final Widget placeholder;
+
+  /// Defines if the player will start in fullscreen when play is pressed
+  final bool fullScreenByDefault;
+
+  /// Defines if the player will sleep in fullscreen or not
+  final bool allowedScreenSleep;
+
+  /// Defines if the controls should be for live stream video
+  final bool isLive;
+
+  Future _initialize() async {
+    await value.videoPlayerController.setLooping(looping);
+
+    if (autoInitialize || autoPlay) {
+      await value.videoPlayerController.initialize();
+    }
+
+    if (autoPlay) {
+      if (fullScreenByDefault) {
+        enterFullscreen();
+      }
+
+      await value.videoPlayerController.play();
+    }
+
+    if (startAt != null) {
+      await value.videoPlayerController.seekTo(startAt);
+    }
+
+    if (fullScreenByDefault) {
+      value.videoPlayerController.addListener(() async {
+        if (await value.videoPlayerController.value.isPlaying &&
+            !value.isFullScreen) {
+          enterFullscreen();
+        }
+      });
+    }
+  }
+
+  void enterFullscreen() {
+    value = value.copyWith(isFullScreen: true);
+  }
+
+  void exitFullscreen() {
+    value = value.copyWith(isFullScreen: false);
+  }
+
+  void toggleFullscreen() {
+    value = value.copyWith(isFullScreen: !value.isFullScreen);
+  }
+
+  void play() {
+    value.videoPlayerController.play();
+  }
+
+  void pause() {
+    value.videoPlayerController.pause();
+  }
+
+  // TODO: Do we really need the ability to change the controller?
+  set videoPlayerController(VideoPlayerController controller) {
```

Comment:
I would say probably not -- if someone wants to show a different video, I think it makes sense to dispose of the ChewieController and create a new one.

Things like AspectRatio, etc might not be handled correctly for different videos

Reply:
Sounds good to me.

Up until now this was being done in `Chewie`: https://github.com/brianegan/chewie/blob/b135a466622daf080282b756cf6749844ed05632/lib/src/chewie_player.dart#L180-L185

Maybe we should adapt the example to show the proper disposing of both of the controllers. Or is there a better way we can help the user?

Reply:
I think we should generally encourage folks to dispose of both separately, because it seems like there are a lot of cases where folks do NOT want Chewie to do that automatically for them (and where it causes bugs).

What are your thoughts?

Reply:
Let's leave it to the user for the moment.